chore: clean up test setup and address #48 comments

kleyow opened the pull request.

partiallyordered commented on integration_test/tests/src/tests/HubManager.test.ts:
```diff
@@ -5,6 +5,9 @@ import { HubManagerPage, DFSPRow } from '../page-objects/pages/HubManager';
 import chance from 'chance';
 import { Selector } from 'testcafe';
 
+// NOTE: There are added `t.wait` functions on `t.clicks` to give the UI time to
+//       update. This is to account for observed flakiness with test-cafe
+//       tests.
```
It would be useful to elaborate slightly on flakiness, if possible. Something like "it was observed that tcafe clicked the wrong element" or "tcafe seemed to click before the page loaded" or "tcafe seemed to fail the test too quickly when it didn't detect the element". That said, I know it's sometimes very difficult to elaborate on flakiness, because if you knew what the problem was, you might stand a chance of fixing it.

kleyow replied:
done.

partiallyordered reviewed (commented):
Some minors

kleyow commented on integration_test/manifests/connection-manager-api/kustomization.yaml:
```diff
@@ -1,51 +1,81 @@
+      - VAULT_ROLE_ID_FILE=/vault/secrets/role-id
+      - VAULT_ROLE_SECRET_ID_FILE=/vault/secrets/secret-id
+      - VAULT_ENDPOINT=http://vault:8200
+      - VAULT_PKI_BASE_DOMAIN=example.com
```
done. the cert is expired but that was the one I was given for testing.

partiallyordered reviewed (approved):
Looks good. A question: are we sure it's okay to commit `cluster-keys.json` (particularly the root token) and this cert? (I see you did mention the cert is expired..)

If you accidentally added some real values, check this out: https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/removing-sensitive-data-from-a-repository

Though the best thing to do would just be to regenerate those values, in particular the root token.

I've approved assuming none of this is sensitive (and besides, merging it isn't going to change the reality of its existence on the internet..)

kleyow commented:
> Looks good. A question: are we sure it's okay to commit cluster-keys.json (particularly the root token) and this cert? (I see you did mention the cert is expired..)

`cluster-keys.json` is just emphemeral from vault and it can be commited but shouldn't because it should be only for local testing. Removed and added to gitignore.

None of this is sensative afaik.
